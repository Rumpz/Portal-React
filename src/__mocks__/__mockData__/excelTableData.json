{"headers":["MERGE2","TIPO_SINTOMA2","AREA_SINTOMA2","SUB_AREA_SINTOMA2","Incidência do Sintoma2","Obs2","OBSXX2","OBSvvv4"],"data":[["APPS TV IRIS | APP COM ERRO | MODEM NÃO REGISTA","APPS TV IRIS","APP COM ERRO","MODEM NÃO REGISTA","TV","-","-","-"],["APPS TV IRIS | ERRO NO ACESSO | DIAGNÓSTICO INCONCLUSIVO","APPS TV IRIS","ERRO NO ACESSO","DIAGNÓSTICO INCONCLUSIVO","TV","-","-","-"],["APPS TV IRIS | INFORMAÇÃO INDISPONÍVEL | DIAGNÓSTICO INCONCLUSIVO","APPS TV IRIS","INFORMAÇÃO INDISPONÍVEL","DIAGNÓSTICO INCONCLUSIVO","TV","-","-","-"],["APPS TV IRIS | ERRO NO ACESSO | PROBLEMA CONHECIDO","APPS TV IRIS","ERRO NO ACESSO","PROBLEMA CONHECIDO","TV","-","-","-"],["APPS TV IRIS | APP COM ERRO | MODEM NÃO REGISTA","APPS TV IRIS","APP COM ERRO","MODEM NÃO REGISTA","TV","-","-","-"],["APPS TV IRIS | ERRO NO ACESSO | DIAGNÓSTICO INCONCLUSIVO","APPS TV IRIS","ERRO NO ACESSO","DIAGNÓSTICO INCONCLUSIVO","TV","-","-","-"],["APPS TV IRIS | INFORMAÇÃO INDISPONÍVEL | DIAGNÓSTICO INCONCLUSIVO","APPS TV IRIS","INFORMAÇÃO INDISPONÍVEL","DIAGNÓSTICO INCONCLUSIVO","TV","-","-","-"],["APPS TV IRIS | ERRO NO ACESSO | PROBLEMA CONHECIDO","APPS TV IRIS","ERRO NO ACESSO","PROBLEMA CONHECIDO","TV","-","-","-"],["APPS TV IRIS | APP COM ERRO | MODEM NÃO REGISTA","APPS TV IRIS","APP COM ERRO","MODEM NÃO REGISTA","TV","-","-","-"],["APPS TV IRIS | ERRO NO ACESSO | DIAGNÓSTICO INCONCLUSIVO","APPS TV IRIS","ERRO NO ACESSO","DIAGNÓSTICO INCONCLUSIVO","TV","-","-","-"],["APPS TV IRIS | INFORMAÇÃO INDISPONÍVEL | DIAGNÓSTICO INCONCLUSIVO","APPS TV IRIS","INFORMAÇÃO INDISPONÍVEL","DIAGNÓSTICO INCONCLUSIVO","TV","-","-","-"],["APPS TV IRIS | ERRO NO ACESSO | PROBLEMA CONHECIDO","APPS TV IRIS","ERRO NO ACESSO","PROBLEMA CONHECIDO","TV","-","-","-"],["APPS TV IRIS | APP COM ERRO | MODEM NÃO REGISTA","APPS TV IRIS","APP COM ERRO","MODEM NÃO REGISTA","TV","-","-","-"],["APPS TV IRIS | ERRO NO ACESSO | DIAGNÓSTICO INCONCLUSIVO","APPS TV IRIS","ERRO NO ACESSO","DIAGNÓSTICO INCONCLUSIVO","TV","-","-","-"],["APPS TV IRIS | INFORMAÇÃO INDISPONÍVEL | DIAGNÓSTICO INCONCLUSIVO","APPS TV IRIS","INFORMAÇÃO INDISPONÍVEL","DIAGNÓSTICO INCONCLUSIVO","TV","-","-","-"],["APPS TV IRIS | ERRO NO ACESSO | PROBLEMA CONHECIDO","APPS TV IRIS","ERRO NO ACESSO","PROBLEMA CONHECIDO","TV","-","-","-"]]}