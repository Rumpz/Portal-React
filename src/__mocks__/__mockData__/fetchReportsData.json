[
  {
    "id":1,
    "codigo_coisas":"1-CVERTTV",
    "outros_codigos":"Done"
  },
  {
    "id":2,
    "codigo_coisas":"1-XXXXXXX",
    "outros_codigos":"Almost Done"
  },
  {
    "id":3,
    "codigo_coisas":"1-YYYYYYY",
    "outros_codigos":"Doing it"
  }
]