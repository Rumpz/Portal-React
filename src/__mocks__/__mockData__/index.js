module.exports = {
  fetchReportsData: require('./fetchReportsData'),
  excelTableData: require('./excelTableData')
};
