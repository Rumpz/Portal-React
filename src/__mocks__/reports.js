/* // __mocks__/request.js
import { fetchReportsData } from './__mockData__';

export default function fetchReports(url) {
  return new Promise((resolve, reject) => {
    process.nextTick(
      () =>
        url === 'reporting'
          ? resolve(fetchReportsData)
          : reject({
              error: 'Status 404 Not Found',
            }),
    );
  });
} */

export default {
  get: jest.fn(() => Promise.resolve({ data: {} }))
};