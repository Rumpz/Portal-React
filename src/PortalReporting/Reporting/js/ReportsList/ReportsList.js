import React from 'react';
import './css/reports.css';
import Checkbox from '@material-ui/core/Checkbox';

let filterValues = [];
const ReportsList = (props) => {
  return (
    <div style={{marginTop: '10px', marginBottom: '10px'}} className='table table-hover'>
      <h4>Filtrar por subcategoria</h4>
      <ul style={{display: 'inline-flex'}}>
        {props.subCat.map((el, index) => {
          return (
            <li key={el + index}>
              <label>{el}</label>
              <Checkbox
                id={`checkBox-${el}`}
                value={el}
                onChange={(event, checked) => {
                  return checked
                    ? filterValues.push(event.target.value)
                    : filterValues.splice(filterValues.indexOf(event.target.value));
                }}
              />
            </li>
          );
        })}
        <button
          id='filterButton'
          className='btn btn-warning'
          onClick={props.actions.checkSubCat.bind(null, filterValues)}>
          <i className='fa fa-search' />Pesquisar
        </button>
      </ul>
      <strong><p id='filterP'>{props.errMsg}</p></strong>
      <table>
        <Header arrValues={props.header} />
        <Body body={props.values} actions={props.actions} style={props.style} />
      </table>
    </div>
  );
};

const Header = ({arrValues}) => {
  let thVals = arrValues.map((value, idx) => {
    return <th key={idx}>{value}</th>;
  });
  return (
    <thead>
      <tr>{thVals}</tr>
    </thead>
  );
};

const Body = ({body, actions, style}) => {
  let row = body.map((value, idx) => {
    return <Row style={style} key={idx} value={value} actions={actions} />;
  });
  return (
    <tbody>
      {row}
    </tbody>
  );
};

const Row = ({value, actions, style}) => {
  let tRows = value.values.map((element, idx) => {
    return <Cell key={idx} value={element}><button className='icon-home icon-4x'>..</button></Cell>;
  });
  return (
    <tr onClick={actions ? actions.row.bind(null, value) : null}>{tRows}</tr>
  );
};

const Cell = ({value, style}) => {
  return <td style={style} >{value}</td>;
};

export default ReportsList;
