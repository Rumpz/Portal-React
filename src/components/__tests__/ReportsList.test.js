
import EnzymeAdapter from 'enzyme-adapter-react-16';
import ShallowRenderer from 'react-test-renderer/shallow';
import Enzyme from 'enzyme';
import React from 'react';
import ReportsList from '../Reports/ReportsList';
import { fetchReportsData } from '../../__mocks__/__mockData__';
// This sets up the adapter to be used by Enzyme
Enzyme.configure({ adapter: new EnzymeAdapter() });

// Test Reports Component Render
// It should render an unordered list with the given data drray
it('renders reports list correctly', async () => {
  const renderer = new ShallowRenderer();
  renderer.render(<ReportsList data={fetchReportsData} />);
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});
