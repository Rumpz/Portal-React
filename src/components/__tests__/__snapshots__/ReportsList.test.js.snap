// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders reports list correctly 1`] = `
<div
  className="table table-hover"
  style={
    Object {
      "marginBottom": "10px",
      "marginTop": "10px",
    }
  }
>
  <h4>
    Filtrar por subcategoria
  </h4>
  <ul
    style={
      Object {
        "display": "inline-flex",
      }
    }
  >
    <li>
      <button
        className="btn btn-default"
        style={
          Object {
            "marginLeft": "12px",
            "width": "200px",
          }
        }
      >
        KC
      </button>
    </li>
    <li>
      <button
        className="btn btn-default"
        style={
          Object {
            "marginLeft": "12px",
            "width": "200px",
          }
        }
      >
        KC
      </button>
    </li>
    <li>
      <button
        className="btn btn-default"
        style={
          Object {
            "marginLeft": "12px",
            "width": "200px",
          }
        }
      >
        KC
      </button>
    </li>
  </ul>
  <table>
    <Header
      arrValues={undefined}
    />
    <Body
      actions={undefined}
      body={undefined}
      style={undefined}
    />
  </table>
</div>
`;
